apps/hash-api/src/graph/ontology/primitive/entity-type.ts

```diff
@@ -38,6 +39,46 @@ import { NotFoundError } from "../../../lib/error";
 import { ImpureGraphFunction } from "../..";
 import { getNamespaceOfAccountOwner } from "./util";
 
+export const getEntityTypeAuthorizationRelationships: ImpureGraphFunction<
+  { entityTypeId: VersionedUrl },
+  Promise<EntityTypeAuthorizationRelationship[]>
+> = async ({ graphApi }, { actorId }, params) =>
+  graphApi
+    .getEntityTypeAuthorizationRelationships(actorId, params.entityTypeId)
+    .then(({ data }) =>
+      data.map(
+        (relationship) =>
+          ({
+            resource: { kind: "entityType", resourceId: params.entityTypeId },
+            ...relationship,
+          }) as EntityTypeAuthorizationRelationship,
```

Comment:
Is there a way we can avoid casting here? This could lead to us forgetting to update something in the Node API if the return type changes.

Reply:
This was raised before when I implemented this for entities [a while back](https://github.com/hashintel/hash/pull/3403/files). We had a quite lengthy discussion about this in standup and it's not possible to avoid the casts as we need to cast from `string` to `AccountId`. The whole point of the cast is to make it more safe.

In this PR I only moved the function to the top so I can use it later.

Reply:
We can change that but I prefer to not doing this as a drive-by in this PR as this PR is unrelated.

Reply:
Okay let's create a follow-up then, I'm keen to avoid adding more casts where possible given the difficulties we had finding and removing the various `Subgraph` casts